tekton/pipeline_run.go

```diff
@@ -123,13 +129,34 @@ func (r *ReleasePipelineRun) WithOwner(release *v1alpha1.Release) *ReleasePipeli
 
 // WithReleaseLabels adds Release name, namespace, and workspace as labels to the release PipelineRun.
 func (r *ReleasePipelineRun) WithReleaseLabels(releaseName, releaseNamespace, releaseWorkspace string) *ReleasePipelineRun {
-	r.ObjectMeta.Labels = map[string]string{
+	releaseLabels := map[string]string{
 		PipelinesTypeLabel:    PipelineTypeRelease,
 		ReleaseNameLabel:      releaseName,
 		ReleaseNamespaceLabel: releaseNamespace,
 		// PipelineRun does not allow labels with : in the value, which KCP workspaces have
 		ReleaseWorkspaceLabel: strings.ReplaceAll(releaseWorkspace, ":", "__"),
 	}
+	// makes sure previously added labels are kept
```

Comment:
If you use a loop like you are doing, other values will always be kept. You only override keys, so I don't think you need the if block.

Reply:
@davidmogar It seems I still need the `if` because Go (for some reason) does not allow me to add new fields to the map if it was initialized without any. It panics with `assignment to entry in nil map` as for instance:

```
	pipelineRun.ObjectMeta.Labels = make(map[string]string)
	pipelineRun.ObjectMeta.Labels["foor"] = "bar"  _// panics with assignment to entry in nil map_
```
or
```
	pipelineRun.ObjectMeta.Labels = map[string]string{}
	pipelineRun.ObjectMeta.Labels["foor"] = "bar"  _// panics with assignment to entry in nil map_
```

But it works  if I initialize it with any `key`, `value` item:

```
	pipelineRun.ObjectMeta.Labels = map[string]string{ "foo": "bar"}
	pipelineRun.ObjectMeta.Labels["a"] = "b"  _// works as expected_
```

It would still require to make sure we initialize the `Labels` field properly in  the `With...Labels` functions, hence checking if `...Labels` is _nil_ and then creating the map content with the proper items on it.

So now I agree even more with your idea to consolidate both in just one function, so we avoid that and check this just once.

Reply:
Just an update: It was caused by the `k8sclient.Create` in the `BeforeEach` routine of the test. It seems it resets the empty `Labels` map making necessary to re-initializing it again.